app/src/main/java/ch/epfl/skysync/viewmodel/UserManagementViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package ch.epfl.skysync.viewmodel
+
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.compose.viewModel
+import ch.epfl.skysync.Repository
+import ch.epfl.skysync.components.SnackbarManager
+import ch.epfl.skysync.models.user.TempUser
+import ch.epfl.skysync.models.user.User
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/** viewmodel for the user management */
+class UserManagementViewModel(
+    val repository: Repository,
+    val userId: String?,
+) : ViewModel() {
+  companion object {
+    @Composable
+    fun createViewModel(
+        repository: Repository,
+        userId: String?,
+    ): UserManagementViewModel {
+      return viewModel<UserManagementViewModel>(
+          factory =
+              object : ViewModelProvider.Factory {
+                override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                  return UserManagementViewModel(repository, userId) as T
+                }
+              })
+    }
+  }
+
+  private val _allUsers: MutableStateFlow<List<User>> = MutableStateFlow(emptyList())
+  private val _selectedUser: MutableStateFlow<User?> = MutableStateFlow(null)
+
+  val allUsers = _allUsers.asStateFlow()
+  val selectedUser = _selectedUser.asStateFlow()
+
+  /** Refreshes the data of the viewmodel */
+  fun refresh() {
+    refreshAllUsers()
+    refreshSelectedUser()
+  }
+
+  /** Refreshes the data of all users */
+  private fun refreshAllUsers() {
+    viewModelScope.launch {
+      _allUsers.value = repository.userTable.getAll(onError = { onError(it) })
+      Log.d("UserManagementViewModel", "All users loaded")
+    }
+  }
+
+  /** Refreshes the data of the selected user */
+  private fun refreshSelectedUser() {
+    viewModelScope.launch {
+      _selectedUser.value = repository.userTable.get(userId!!, onError = { onError(it) })
+      Log.d("UserManagementViewModel", "Selected user loaded")
+    }
+  }
+
+  /** Adds a user to the database */
+  fun addUser(tmpUser: TempUser) {
```

Comment:
Here maybe naming the function `createUser` and stating in the documentation that you create a temporary user in the `temp-user` table would make this clearer, because right now `addUser` doesn't add a user in the user table.

Reply:
I changed it